### PR TITLE
[JENKINS-44860] Allow usernameVariable to be omitted to bypass masking

### PR DESCRIPTION
**Pull request by jglick**
Less intrusive resolution to [JENKINS-44860](https://issues.jenkins-ci.org/browse/JENKINS-44860) compared to #44 / #50.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/UsernamePasswordMultiBinding.java [hunk 1]
```diff
@@ -45,22 +45,26 @@
...
+import org.kohsuke.stapler.DataBoundSetter;
 
 public class UsernamePasswordMultiBinding extends MultiBinding<StandardUsernamePasswordCredentials> {
 
-    private final String usernameVariable;
+    private String usernameVariable;
     private final String passwordVariable;
 
-    @DataBoundConstructor public UsernamePasswordMultiBinding(String usernameVariable, String passwordVariable, String credentialsId) {
+    @DataBoundConstructor public UsernamePasswordMultiBinding(String passwordVariable, String credentialsId) {
```
[the remark this answers is not in the document]
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/UsernamePasswordMultiBinding.java]: [Apparently so](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/7502d14aa4db891302abf48a2c7de7c9ced97f95/pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/UsernamePasswordHandler.java#L49), sigh.

**Comment by valentin-nasta** [comment]
Hey guys, any update on this PR?

**Comment by jglick** [comment]
Not recently. Whatever approach is used needs to also be integrated with `pipeline-model-definition-plugin`.

**Comment by Herr-Sepp** [comment]
I also have the problem that with common usernames (e.g. jenkins) my whole log becomes badly readable.
It is not only the other log places that are affected, I even want the user to be explicitly written in the log for good traceability.

This function is also constantly requested under [JENKINS-44860](https://issues.jenkins-ci.org/browse/JENKINS-44860).

I do not recognize any active PR for [pipeline-model-definition-plugin](https://github.com/jenkinsci/pipeline-model-definition-plugin/pulls) with this problem in mind. Is there where help is needed?

What currently holds back this PR?

**Comment by jglick** [comment]
> I do not recognize any active PR for [pipeline-model-definition-plugin](https://github.com/jenkinsci/pipeline-model-definition-plugin/pulls) with this problem in mind.

Right, one would need to be written.

**Review by Jankyboy — APPROVED** [review]

**Comment by jglick** [comment]
#105 by @car-roll is likely to clash with any of three PRs associated with this feature. Needs to be reconsidered after that is done. My general criteria for any PR like this:

* Should use a defined API, not a one-off trick (e.g., no casting to an implementation class).
* Should be opt-in, so behavior of existing projects is not affected.

**Comment by jglick** [comment]
Planning to go with the approach in #50 except making it opt-in.

**Comment by jglick** [comment]
(see #131)

**Pull request closed without merging**